pkg/osquery/table/email_addresses.go

```diff
@@ -16,32 +16,43 @@ func EmailAddresses(client *osquery.ExtensionManagerClient) *table.Plugin {
 		table.TextColumn("email"),
 		table.TextColumn("domain"),
 	}
-	return table.NewPlugin("kolide_email_addresses", columns, generateEmailAddresses(client))
+	t := &emailAddressesTable{
+		onePasswordAccountConfig: &onePasswordAccountConfig{client: client},
+	}
+	return table.NewPlugin("kolide_email_addresses", columns, t.generateEmailAddresses)
 }
 
-func generateEmailAddresses(client *osquery.ExtensionManagerClient) table.GenerateFunc {
-	return func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
-		results := []map[string]string{}
+type emailAddressesTable struct {
+	onePasswordAccountConfig *onePasswordAccountConfig
+}
 
-		for _, stateFilePath := range findChromeStateFiles() {
-			fileContent, err := ioutil.ReadFile(stateFilePath)
-			if err != nil {
-				return nil, errors.Wrapf(err, "could not read file %s", stateFilePath)
-			}
+func (t *emailAddressesTable) generateEmailAddresses(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	results := []map[string]string{}
 
-			var parsedStateFileContent chromeLocalStateFile
-			if err := json.Unmarshal(fileContent, &parsedStateFileContent); err != nil {
-				return nil, errors.Wrap(err, "could not unmarshal json file")
-			}
+	// add results from chrome profiles
+	for _, stateFilePath := range findChromeStateFiles() {
```

Comment:
I would feel better if this were abstracted into a `generate` function like the 1password portion.